### PR TITLE
Added a noopcachemanager verification for user resources tests

### LINKED ISSUE #9901: UserResourceIT tests crashing when using memcached
##### **Overview of the issue**

When generating an application using memcached, the caching is faked (it uses NoOpCache instead) when using dev profile. This is because spring devtools cannot be used with memcached.

This makes UserResourceIT tests crash.

Trace: [see link](https://dev.azure.com/phamjerome/jhipster-daily-builds/_build/results?buildId=760)

##### **Motivation for or Use Case**

Tests should not crash just after generating the project.

##### **Reproduce the error**

```
application {
 config {
   baseName store,
   applicationType gateway,
   packageName com.jhipster.demo.store,
   serviceDiscoveryType eureka,
   authenticationType jwt,
   prodDatabaseType mysql,
   cacheProvider memcached,
   buildTool maven,
   clientFramework angularX,
   enableSwaggerCodegen true,
   testFrameworks [protractor]
 }
 entities *
}

application {
 config {
   baseName crm,
   applicationType microservice,
   packageName com.jhipster.demo.crm,
   serviceDiscoveryType eureka,
   authenticationType jwt,
   
   prodDatabaseType postgresql,
   buildTool maven,
   searchEngine elasticsearch,
   serverPort 8081,
   skipUserManagement true
 }
 entities * except Invoice, Shipment
}


entity Product {
   name String required
   description String
   price BigDecimal required min(0)
   size Size required
   image ImageBlob
}
enum Size {
   S, M, L, XL, XXL
}
entity Customer {
   firstName String required
   lastName String required
   gender Gender required
   email String required pattern(/^[^@\s]+@[^@\s]+\.[^@\s]+$/)
   phone String required
   addressLine1 String required
   addressLine2 String
   city String required
   country String required
}
enum Gender {
   MALE, FEMALE, OTHER
}
entity ProductOrder {
   placedDate Instant required
   status OrderStatus required
   code String required
   invoiceId String
}
enum OrderStatus {
   COMPLETED, PENDING, CANCELLED
}
entity OrderItem {
   quantity Integer required min(0)
   totalPrice BigDecimal required min(0)
   status OrderItemStatus required
}
enum OrderItemStatus {
   AVAILABLE, OUT_OF_STOCK, BACK_ORDER
}
relationship ManyToOne {
OrderItem{product(name) required} to Product
}
relationship OneToMany {
  Customer{order} to ProductOrder{customer(email) required},
  ProductOrder{orderItem} to OrderItem{order(code) required}
}

service Product, Customer, ProductOrder, OrderItem with serviceClass
paginate Product, Customer, ProductOrder, OrderItem with pagination

/* Entities for Invoice microservice */
entity Invoice {
   code String required
   date Instant required
   details String
   status InvoiceStatus required
   paymentMethod PaymentMethod required
   paymentDate Instant required
   paymentAmount BigDecimal required
}
enum InvoiceStatus {
   PAID, ISSUED, CANCELLED
}
entity Shipment {
   trackingCode String
   date Instant required
   details String
}
enum PaymentMethod {
   CREDIT_CARD, CASH_ON_DELIVERY, PAYPAL
}
relationship OneToMany {
   Invoice{shipment} to Shipment{invoice(code) required}
}
service Invoice, Shipment with serviceClass
paginate Invoice, Shipment with pagination
microservice Invoice, Shipment with accountancy
microservice Customer, Product, ProductOrder, OrderItem with crm

// will be created under 'docker-compose' folder
deployment {
   deploymentType docker-compose,
   appsFolders [store, crm],
   dockerRepositoryName "jeromepham"
   monitoring no,
   serviceDiscoveryType eureka
}

// will be created under 'kubernetes' folder
deployment {
   deploymentType kubernetes,
   appsFolders [store, crm]
   dockerRepositoryName "jeromepham",
   gatewayType zuul,
   monitoring no,
   serviceDiscoveryType eureka
   kubernetesNamespace jhipster,
   kubernetesServiceType LoadBalancer,
   istio false
}
```

generate the application using the above jdl configuration. go into store folder and ./mvnw -Pprod verify jib:dockerBuild

##### **Related issues**

/

##### **Suggest a Fix**

Check if the NoOpCache is used to skip the assert.

##### **JHipster Version(s)**

master version of the current date. (6.1.0)

##### **JHipster configuration**

```INFO! Using JHipster version installed globally
INFO! Executing jhipster:info
INFO! Options: from-cli: true
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
integ-kubernetes@1.0.0 /home/ippon/workspace/integ-kubernetes
└── (empty)

```

cat: no such file or directory: .yo-rc.json

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>

</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

openjdk version "12.0.1" 2019-04-16
OpenJDK Runtime Environment (build 12.0.1+12)
OpenJDK 64-Bit Server VM (build 12.0.1+12, mixed mode, sharing)

git version 2.17.1

node: v10.16.0

npm: 6.9.0

yeoman: 2.0.5

yarn: 1.13.0

Docker version 18.09.6, build 481bc77

docker-compose version 1.23.2, build 1110ad01

INFO! Congratulations, JHipster execution is complete!
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

See JDL above

##### **Browsers and Operating System**

ubuntu 18.04

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by Hawkurane**
This PR is to make sure when the memcached is disabled, the tests are taking note of it.

Fixes #9901 

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs [hunk 1]
```diff
@@ -519,7 +522,7 @@ public class UserResourceIT <% if (databaseType === 'cassandra') { %>extends Abs
         <%_ if (searchEngine === 'elasticsearch') { _%>
         mockUserSearchRepository.save(user);
         <%_ } _%>
-        <%_ if (cacheManagerIsAvailable === true) { _%>
+        <%_ if (cacheManagerIsAvailable === true) {  _%>
```
Comment (pvliss) [inline generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs]: Really minor, but the extra space is not needed :)

File: generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs [hunk 2]
```diff
@@ -552,8 +555,11 @@ public class UserResourceIT <% if (databaseType === 'cassandra') { %>extends Abs
             <%_ } _%>
             .jsonPath("$.langKey").isEqualTo(DEFAULT_LANGKEY);
         <%_ } _%>
-        <%_ if (cacheManagerIsAvailable === true) { _%>
 
+        <%_ if (cacheProvider === 'memcached') {  _%>
+        if(!(cacheManager instanceof NoOpCacheManager))
```
Comment (pascalgrimaud) [inline generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs]: format, need space, need { and }

Comment (pvliss) [reply generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs]: Now, that's a good eye :) How, did I miss the missing {} :hushed:

Comment (Hawkurane) [reply generators/server/templates/src/test/java/package/web/rest/UserResourceIT.java.ejs]: Indeed I used my own style of coding there which isn't that popular.
I've tried fixing it but I am unsure if it fits the standart

**Comment by jdubois** [comment]
Thanks @Hawkurane !

**Pull request merged**